operator/slotticker/slotticker_test.go

```diff
@@ -178,3 +179,54 @@ func TestSlotSkipping(t *testing.T) {
 		}
 	}
 }
+
+type mockTimerProvider struct {
+	C chan time.Time
+}
+
+func (mtp *mockTimerProvider) NewTimer(d time.Duration) *time.Timer {
+	// Create a timer with a large duration and immediately stop it.
+	// This is to create a properly initialized timer.
+	t := time.NewTimer(time.Hour)
+	// t.Stop()
+
+	// Replace the timer's channel with our mock channel.
+	t.C = mtp.C
+	return t
+}
+
+func TestDoubleTickWarning(t *testing.T) {
+	// Create a mock timer channel
+	mockTimerChan := make(chan time.Time, 2)
+
+	// Setting up a logger with observer to capture the warning logs
+	core, recorded := observer.New(zap.WarnLevel)
+	logger := zap.New(core)
+
+	// Initialize the slotTicker with the mock timer provider
+	ticker := NewWithCustomTimer(logger, Config{
+		slotDuration: 200 * time.Millisecond,
+		genesisTime:  time.Now(),
+	}, &mockTimerProvider{C: mockTimerChan})
+
+	// Manually fire the timer twice to simulate rapid ticks
+	mockTimerChan <- time.Now()
+	mockTimerChan <- time.Now()
+
+	// Call Next() twice to process the ticks
+	<-ticker.Next()
+	<-ticker.Next()
+
```

Comment:
shouldn't we have also different slots between these calls? I would add a check for this. IMO that's the actual test more than checking the logs.